ui/app/mainui/Handlers/HandlersManager.qml

```diff
@@ -0,0 +1,141 @@
+import QtQuick 2.15
+
+import StatusQ.Core.Utils 0.1 as SQUtils
+
+import shared.stores 1.0 as SharedStores
+import shared.stores.send 1.0 as SharedSendStores
+
+import AppLayouts.stores 1.0 as AppStores
+import AppLayouts.Wallet.stores 1.0 as WalletStores
+import AppLayouts.Chat.stores 1.0 as ChatStores
+
+import AppLayouts.Wallet.popups.buy 1.0
+import AppLayouts.Wallet.popups.swap 1.0
+import utils 1.0
+
+QtObject {
+    id: root
+
+    required property var popupParent
+
+    // Stores definition:
+    required property AppStores.RootStore rootStore
+    required property AppStores.FeatureFlagsStore featureFlagsStore
+
+    required property SharedStores.RootStore sharedRootStore
+    required property SharedStores.CurrenciesStore currencyStore
+    required property SharedStores.NetworksStore networksStore
+    required property SharedSendStores.TransactionStore transactionStore
+
+    required property var/*TODO: Apply strong typing onces its no longer a singleton*/ walletRootStore
+    required property WalletStores.WalletAssetsStore walletAssetsStore
+    required property WalletStores.CollectiblesStore walletCollectiblesStore
+    required property WalletStores.TransactionStoreNew transactionStoreNew
+    required property WalletStores.TokensStore tokensStore
+
+    required property ChatStores.RootStore rootChatStore
+
+    readonly property SwapModalHandler swapModalHandler: SwapModalHandler {
+
+        property SwapInputParamsForm swapFormData: SwapInputParamsForm {}
+
+        function launchSwap() {
+            swapFormData.selectedAccountAddress =
+                    SQUtils.ModelUtils.get(root.walletRootStore.nonWatchAccounts, 0, "address")
+            swapFormData.selectedNetworkChainId =
+                    SQUtils.ModelUtils.getByKey(root.networksStore.activeNetworks, "layer", 1, "chainId")
+            Global.openSwapModalRequested(swapFormData, (popup) => {
+                                              popup.Component.destruction.connect(() => {
+                                                                                      swapFormData.resetFormData()
+                                                                                  })})
+        }
+
+        popupParent: root.popupParent
```

Comment:
also add a TODO here, that the SwapModalHandler like the SendModalHandler needs to have specific apis , properties and function and not just accept stores? I dont think that goes in line with overall approach

Reply:
In the handler level I think it's ok, but in the modal level should be explicit api, yes.